orchagent/vxlanorch.cpp

```diff
@@ -731,41 +1415,280 @@ bool VxlanTunnelOrch::delOperation(const Request& request)
+    EvpnNvoOrch* evpn_orch = gDirectory.get<EvpnNvoOrch*>();
+    std::string remote_vtep;
+
+    auto vtep_ptr = evpn_orch->getEVPNVtep();
+
+    SWSS_LOG_INFO("Delete Tunnelport Enter");
+    if(!vtep_ptr) 
+    {
+      SWSS_LOG_WARN("Unable to find VTEP. tunnelPort=%s",tunnelPort.m_alias.c_str());
+      return;
+    }
+
+    getTunnelDIPFromPort(tunnelPort, remote_vtep);
+
+    //If there are IMR/IP routes to the remote VTEP then ignore this call
+    if(vtep_ptr->getDipTunnelRefCnt(remote_vtep))
+      return;
+
+    // Remove Bridge port and Port objects for this DIP tunnel
+    ret = gPortsOrch->removeBridgePort(tunnelPort);
+    if(!ret) 
+    {
+      SWSS_LOG_ERROR("Remove Bridge port failed for remote = %s fdbcount = %d", 
+                      remote_vtep.c_str(), tunnelPort.m_fdb_count);
+      return;
+    }
+    gPortsOrch->removeTunnel(tunnelPort);
+
+    // Remove DIP Tunnel HW 
+    vtep_ptr->deleteDynamicDIPTunnel(remote_vtep, TNL_SRC_IMR, false);
+    SWSS_LOG_NOTICE("diprefcnt for remote %s = %d",
+                     remote_vtep.c_str(), vtep_ptr->getDipTunnelRefCnt(remote_vtep));
+
+    // Remove SIP Tunnel HW which might be pending on delete
+    vtep_ptr->deletePendingSIPTunnel();
+
+    return ;
+}
+
+void VxlanTunnelOrch::getTunnelName(string& tunnel_portname, string& tunnel_name)
+{
+   tunnel_name = tunnel_portname;
+   tunnel_name.erase(0, sizeof("Port_")-1);
+
+   SWSS_LOG_DEBUG("tunnel name = %s",tunnel_name.c_str());
+
+   return;
+}
+
+#if 0
+void VxlanTunnelOrch::getTunnelPortname(string& dip, string& tunnel_portname)
+{
+   tunnel_port_name = "Port_EVPN_" + dip;
+
+   return;
+}
+#endif
+
+void VxlanTunnelOrch:: getTunnelDIPFromPort(Port& tunnelPort, string& remote_vtep)
+{
+    remote_vtep = tunnelPort.m_alias;
+    remote_vtep.erase(0,sizeof("Port_EVPN_")-1);
+}
+
+
+void VxlanTunnelOrch::updateDbTunnelOperStatus(string tunnel_portname, 
+                                               sai_port_oper_status_t status)
+{
+   std::vector<FieldValueTuple> fvVector;
+   std::string tunnel_name;
+
+   if(status == SAI_PORT_OPER_STATUS_UP)
+     fvVector.emplace_back("operstatus", "up");
+   else
+     fvVector.emplace_back("operstatus", "down");
+
+   getTunnelName(tunnel_portname, tunnel_name);
+
+   m_stateVxlanTable.set(tunnel_name, fvVector);
+}
+
+void VxlanTunnelOrch::addRemoveStateTableEntry(string tunnel_name, 
+                                           IpAddress& sip, IpAddress& dip, 
+                                           tunnel_creation_src_t src, bool add)
+
+{
+    std::vector<FieldValueTuple> fvVector, tmpFvVector;
+    //WarmStart::WarmStartState state;
+
+    //WarmStart::getWarmStartState("orchagent",state);
+
+    if(add)
+    {
+      // Add tunnel entry only for non-warmboot case or WB with new tunnel coming up
+      // during WB
+      //if ( (state != WarmStart::INITIALIZED) || 
+      //     !m_stateVxlanTable.get(tunnel_name, tmpFvVector))
+      if(1)
```

Comment:
??

Reply:
This was dependent on sonic-swss-common PR 350 and was hence this way. Now that 350 is checked in 
uncommenting it.